Editor/Scripts/BotManagement/RGBotSynchronizer.cs

```diff
@@ -31,24 +48,295 @@ private static void CreateNewBot()
+                    if (await _this.UpdateBotZipOnRG(botId, md5))
+                    {
+                        ++countUpdated;
+                    }
+                }
+
+                EditorUtility.DisplayProgressBar("Synchronizing Regression Games Bots",
+                    "Downloading new bots from Regression Games", 0.7f);
+
+                // create local record for any new remote bots + pull down their zips
+                var newRemoteBots =
+                    remoteBots.Where(rb => localBots.FirstOrDefault(bnc => bnc.id == rb.id) == null);
+                foreach (var newRemoteBot in newRemoteBots)
+                {
+                    EditorUtility.DisplayProgressBar("Synchronizing Regression Games Bots",
+                        $"Downloading zip for Bot id: {newRemoteBot.id}", 0.75f);
+                    await _this.CreateLocalBotRecordFromRG(newRemoteBot);
+                }
+
+                _this.RemoveZipTempDirectory();
+
+                EditorUtility.DisplayProgressBar("Synchronizing Regression Games Bots", "Refreshing Assets", 0.9f);
+                
+                // force asset refresh for any new bots
+                AssetDatabase.Refresh();
+                
+                RGBotAssetsManager.GetInstance()?.RefreshAvailableBots();
+
+                var endingLocalBots = RGBotAssetsManager.GetInstance()?.GetAvailableBotAssets();
+
+                EditorUtility.DisplayProgressBar("Synchronizing Regression Games Bots", "Complete", 1f);
+
+                RGDebug.LogInfo($"Synchronizing Bots with Regression Games Server... Complete!\r\n" +
+                                $"Created {endingLocalBots.Count - localBots.Count} new local bot entries\r\n" +
+                                $"Created {countCreatedRemote} new remote bot entries\r\n" +
+                                $"Updated {countUpdated} remote bot entries\r\n");
+                                
+
+                if (endingLocalBots.Count != localBots.Count)
+                {
+                    var newBot = endingLocalBots.FirstOrDefault(nb => localBots.FirstOrDefault(lb => lb.id == nb.Bot.id) == null);
+                    if (newBot != null)
+                    {
+                        //If we created any new bot entries, open that directory in the editor
+                        AssetDatabase.OpenAsset(newBot);
+                    }
+                }
+            }
+            finally
+            {
+                EditorUtility.ClearProgressBar();
+            }
+        }
+
+        private async Task CreateLocalBotRecordFromRG(RGBot newRemoteBot)
+        {
+            //Create directory
+            string botFolderPath = _this.CreateBotFolder(newRemoteBot.name, newRemoteBot.id);
+
+            string zipDownloadPath = ZipPathForBot(newRemoteBot.id);
+
+            if (File.Exists(zipDownloadPath))
+            {
+                File.Delete(zipDownloadPath);
```

Comment:
Technically this should never be reached, right? Because if we are creating a local bot from RG, that means that the asset with the same ID didn't exist before?

Reply:
should never be reached.. but.. i had a file open in testing.. and the ziptemp folder failed to delete.. and on the next run one of these files still existed...

Reply:
added comment to code